src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityPrimitiveWaterPump.java

```diff
@@ -0,0 +1,167 @@
+import gregtech.api.metatileentity.MetaTileEntityHolder;
+import gregtech.api.metatileentity.multiblock.IMultiblockAbilityPart;
+import gregtech.api.metatileentity.multiblock.IMultiblockPart;
+import gregtech.api.metatileentity.multiblock.MultiblockAbility;
+import gregtech.api.metatileentity.multiblock.MultiblockControllerBase;
+import gregtech.api.multiblock.BlockPattern;
+import gregtech.api.multiblock.BlockWorldState;
+import gregtech.api.multiblock.FactoryBlockPattern;
+import gregtech.api.multiblock.PatternMatchContext;
+import gregtech.api.render.ICubeRenderer;
+import gregtech.api.render.OrientedOverlayRenderer;
+import gregtech.api.render.Textures;
+import gregtech.api.unification.material.Materials;
+import gregtech.common.blocks.BlockSteamCasing;
+import gregtech.common.blocks.MetaBlocks;
+import gregtech.common.metatileentities.electric.multiblockpart.MetaTileEntityFluidHatch;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.biome.*;
+import net.minecraftforge.fluids.FluidTank;
+import net.minecraftforge.fluids.IFluidTank;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class MetaTileEntityPrimitiveWaterPump extends MultiblockControllerBase {
+
+    private IFluidTank waterTank;
+    private int biomeModifier = 0;
+    private int hatchModifier = 0;
+
+    public MetaTileEntityPrimitiveWaterPump(ResourceLocation metaTileEntityId) {
+        super(metaTileEntityId);
+        resetTileAbilities();
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder holder) {
+        return new MetaTileEntityPrimitiveWaterPump(metaTileEntityId);
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        if (getOffsetTimer() % 20 == 0 && !getWorld().isRemote && isStructureFormed()) {
+            if (biomeModifier == 0) {
+                biomeModifier = getAmountForBiome(getWorld().getBiome(getPos()));
+            }
+            waterTank.fill(Materials.Water.getFluid(biomeModifier * hatchModifier), true);
+        }
+    }
+
+    private static int getAmountForBiome(Biome biome) {
+        Class<? extends Biome> biomeClass = biome.getBiomeClass();
+        if (biomeClass == BiomeOcean.class || biomeClass == BiomeRiver.class) {
+            return 1000;
+        } else if (biomeClass == BiomeSwamp.class) {
+            return 800;
+        } else if (biomeClass == BiomeJungle.class) {
+            return 350;
+        } else if (biomeClass == BiomeSnow.class) {
+            return 300;
+        } else if (biomeClass == BiomePlains.class || biomeClass == BiomeForest.class) {
+            return 250;
+        } else if (biomeClass == BiomeTaiga.class) {
+            return 175;
+        } else if (biomeClass == BiomeBeach.class) {
+            return 170;
+        } else {
+            return 100;
+        }
+    }
```

Comment:
Is it possible to do this based off of Biome Temperature? This would give it more compat with other biome mods that utilize the biome temperature system, like Arid or Temperate, etc.